client/src/services/authenticationService.js

```diff
@@ -0,0 +1,92 @@
+import { BehaviorSubject } from 'rxjs';
+import { useSnackbar } from 'notistack';
+
+import useHandleResponse from '../utilities/handleResponse';
+
+const currentUserSubject = new BehaviorSubject(
+    JSON.parse(localStorage.getItem('currentUser'))
+);
+
+export const authenticationService = {
+    logout,
+    currentUser: currentUserSubject.asObservable(),
+    get currentUserValue() {
+        return currentUserSubject.value;
+    },
+};
+
+export function useLogin() {
+    const { enqueueSnackbar } = useSnackbar();
+    const handleResponse = useHandleResponse();
+
+    const login = (email, password) => {
+        const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ email, password }),
+        };
+
+        return fetch(
+            `${process.env.REACT_APP_API_URL}/api/users/login`,
```

Comment:
For API calls with our setup, you don't need to add the API_URL to .env or to specify it because of the proxy set up in client/package.json. Instead you can just do fetch('/api/users/login') and it'll use hte proxy to route to the back-end